tests/integration/transactions/test_amm_clawback.py

```diff
@@ -0,0 +1,63 @@
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import (
+    create_amm_pool_async,
+    sign_and_reliable_submission_async,
+    test_async_and_sync,
+)
+from xrpl.models.amounts.issued_currency_amount import IssuedCurrencyAmount
+from xrpl.models.currencies.issued_currency import IssuedCurrency
+from xrpl.models.currencies.xrp import XRP
+from xrpl.models.transactions import AMMDeposit
+from xrpl.models.transactions.amm_clawback import AMMClawback
+from xrpl.models.transactions.amm_deposit import AMMDepositFlag
+
+
+class TestAMMClawback(IntegrationTestCase):
+    @test_async_and_sync(globals())
+    async def test_positive_workflow(self, client):
+        amm_pool = await create_amm_pool_async(
+            client, issuer_account_enable_trustline_clawback=True
+        )
+
+        # Asset-1 is XRP, Asset-2 is an IssuedCurrency titled "USD"
+        # The Issuer of Asset-2 is the issuer_wallet
+        # For the purposes of this test, the issuer_wallet has set the
+        # Allow Trust Line Clawback flag
+        issuer_wallet = amm_pool["issuer_wallet"]
+        holder_wallet = amm_pool["lp_wallet"]
+
+        # "holder" account deposits both assets into the AMM pool
+        # Deposit assets into AMM pool
+        amm_deposit = AMMDeposit(
+            account=holder_wallet.address,
+            asset=IssuedCurrency(
+                currency="USD",
+                issuer=issuer_wallet.address,
+            ),
+            asset2=XRP(),
```

Comment:
You can reference `asset` and `asset2` from `amm_pool`.

Reply:
The first `asset` cannot be XRP. rippled will throw a `temMALFORMED` error. However, `asset2` can be XRP. Source: https://github.com/XRPLF/rippled/blob/fb3713bc2507d577aebe56fbc38872346811d76d/src/xrpld/app/tx/detail/AMMClawback.cpp#L63

The `create_amm_pool_async` utility method returns `asset: XRP()`. Hence, I can't directly use this value for this test.

If I refactor this method, I'm not sure if other AMM tests break.

Reply:
Ah yes, you're right. Disregard my comment then.